File: webapp/client/lib/memberColor.ts
const chipColors = [
  "primary",
  "secondary",
  "success",
  "warning",
  "danger",
] as const;

export type ChipColor = (typeof chipColors)[number];

function hashString(str: string): number {
  let hash = 0;
  for (let i = 0; i < str.length; i++) {
    const char = str.charCodeAt(i);
    hash = (hash << 5) - hash + char;
    hash = hash & hash;
  }
  return Math.abs(hash);
}

export function getMemberChipColor(memberName: string): ChipColor {
  const hash = hashString(memberName);
  return chipColors[hash % chipColors.length];
}

File: mobileapp/lib/utils/member_color.dart
import 'package:flutter/material.dart';

/// Get a consistent color for a member based on their name
Color getMemberColor(String name) {
  final hash = name.hashCode.abs();
  final colors = [
    Colors.blue, // primary
    Colors.purple, // secondary
    Colors.green, // success
    Colors.orange, // warning
    Colors.red, // danger
  ];
  return colors[hash % colors.length];
}
